src/nb_enum.cpp

```diff
@@ -150,6 +154,28 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
     if (!t)
         return false;
 
+    if ((t->flags & (uint32_t) type_flags::is_flag_enum) !=0) {
+        auto base = PyObject_GetAttrString((PyObject *)o->ob_type, "__base__");
```

Comment:
These Python API calls return new object references which you must drop (`Py_DECREF`) when you're done using them. Also, this is way too much effort to spend on every individual enum-value conversion from Python to C++. You should do this only if the map lookup in `enum_tbl.rev` fails. If this slower-path conversion succeeds, then you can add to the `enum_tbl.rev` map to save time when converting the same flags value in the future.

Reply:
I'm actually wondering if it makes sense to cache these. With flags, it would seem one can generate 2^n cache entries for n bits, which has the potential of being undesirably large. Maybe flags enums are OK to go on a slow path? Or we somehow restrict the size of the cache?

Reply:
AFAICT, they are already being cached in members of the enum type. This "singleton-izes" so that one can test for flag combinations with `is`.
```
>>> import enum
>>> class Test(enum.Flag):
...   A, B, C, D, E, F, G, H = 1, 2, 4, 8, 16, 32, 64, 128
...
>>> for i in range(256): Test(i)
...
<Test: 0>
<Test.A: 1>
<Test.B: 2>
[etc ...]
>>> len(Test._value2member_map_)
256
```
The existing cache takes, for each referenced combination, one dict entry (like 32 bytes I think?) plus the footprint of the enum value object itself (sys.getsizeof says 48 bytes, but that doesn't include the 4-entry instance dictionary, name string, etc -- it's probably at least 128 bytes total). We're adding one robin_map entry (24 bytes?) in each direction (C++->Py and Py->C++) for each combination that is passed across the language boundary. So the additional cost of caching is something like 30% above what we're already paying just for using `enum.Flag`. That seems reasonable to me.